src/elements/ifc-client-link.ts

```diff
@@ -0,0 +1,53 @@
+import { Client } from '../client';
+const PATH_ATTR = 'path';
+
+/**
+ * A custom element responsible for rendering an anchor element, turning a path
+ * within the current application to a full top-level url.
+ * It is registered as `ifc-client-link` within client.ts
+ */
+
+export default class IfcClientLinkElement extends HTMLElement {
+  private _client: Client;
+
+  /** @internal */
+  constructor(clientInstance: any) {
+    super();
+    this._client = clientInstance;
+  }
+
+  /**
+   * @internal
+   * @inheritdoc
+   */
+  static get observedAttributes() {
+    return [PATH_ATTR];
+  }
+
+  /**
+   * @internal
+   * @inheritdoc
+   */
+  public connectedCallback() {
+    const content = this.innerHTML;
+    let path = this.getAttribute(PATH_ATTR) || 'about:blank';
```

Comment:
One last thing I'd do for this PR is switch the `'about:blank'` default values to just an empty string.  If we append `about:blank` to the client or host URL, we just end up with nonsense, and if we end up with `about:blank` on it's own, we might accidentally send the user to a blank page, which would be a confusing experience.